### PR TITLE
OLS-644: Use version-aligned OCP docs for RAG

### PR DESCRIPTION
**Pull request by syedriko**
## Description

Get OLS to retrieve rag content specific to the OCP version the customer is using

## Type of change

- [ ] Refactor
- [x] New feature
- [ ] Bug fix
- [ ] CVE fix
- [ ] Optimization
- [ ] Documentation Update
- [ ] Configuration Update
- [ ] Bump-up dependent library

## Related Tickets & Documents

- Related Issue #
- Closes # https://issues.redhat.com/browse/OLS-644

## Checklist before requesting a review

- [x] I have performed a self-review of my code.
- [ ] PR has passed all pre-merge test jobs.
- [ ] If it is a core feature, I have added thorough tests.

## Testing
- Please provide detailed steps to perform tests related to this code change.
- How were the fix/results from this change verified? Please provide relevant screenshots or results.

### CODE REVIEW

File: internal/controller/olsconfig_controller.go [hunk 1]
```diff
@@ -97,7 +97,7 @@ type OLSConfigReconcilerOptions struct {
 // +kubebuilder:rbac:groups=monitoring.coreos.com,namespace=openshift-lightspeed,resources=prometheusrules,verbs=get;list;watch;create;update;patch;delete
 
 // clusterversion for checking the openshift cluster version
-// +kubebuilder:rbac:groups=config.openshift.io,resources=clusterversions,verbs=get
+// +kubebuilder:rbac:groups=config.openshift.io,resources=clusterversions,verbs=get;list
```
Comment (raptorsun) [inline internal/controller/olsconfig_controller.go:100]: why `list` verb is requried? we just `get` the object `clusterversion/version` in our code 🤔

**Comment by raptorsun** [comment]
The logics looks good to me :) 
Just a question about the neccessity of `list` verb for the `clusterversion` resource.

File: internal/controller/olsconfig_controller.go (hunk 1 above)
Comment (syedriko) [reply internal/controller/olsconfig_controller.go:100]: I was getting this
```
W0611 01:05:51.062507       1 reflector.go:539] go/pkg/mod/k8s.io/client-go@v0.29.3/tools/cache/reflector.go:229: failed to list *v1.ClusterVersion: clusterversions.config.openshift.io is forbidden: User "system:serviceaccount:openshift-lightspeed:lightspeed-operator-controller-manager" cannot list resource "clusterversions" in API group "config.openshift.io" at the cluster scope
```
in the operator log until I added `list`

Comment (syedriko) [reply internal/controller/olsconfig_controller.go:100]: and more importantly, I coudn't get cluster version

Comment (raptorsun) [reply internal/controller/olsconfig_controller.go:100]: probably it comes from some implementation details in the controller-runtime package. let add the list verb and keep going.

**Comment by syedriko** [comment]
@bparees @xrajesh PTAL with an eye towards approving

**Pull request merged**